specs/_features/epbs/beacon-chain.md

```diff
@@ -0,0 +1,627 @@
+    if not is_parent_block_full(state):
+        return
+
+    withdrawals, partial_withdrawals_count = get_expected_withdrawals(state)
+    state.latest_withdrawals_root = hash_tree_root(withdrawals)
+    for withdrawal in withdrawals:
+        decrease_balance(state, withdrawal.validator_index, withdrawal.amount)
+
+    # Update pending partial withdrawals
+    state.pending_partial_withdrawals = state.pending_partial_withdrawals[partial_withdrawals_count:]
+
+    # Update the next withdrawal index if this block contained withdrawals
+    if len(withdrawals) != 0:
+        latest_withdrawal = withdrawals[-1]
+        state.next_withdrawal_index = WithdrawalIndex(latest_withdrawal.index + 1)
+
+    # Update the next validator index to start the next withdrawal sweep
+    if len(withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD:
+        # Next sweep starts after the latest withdrawal's validator index
+        next_validator_index = ValidatorIndex((withdrawals[-1].validator_index + 1) % len(state.validators))
+        state.next_withdrawal_validator_index = next_validator_index
+    else:
+        # Advance sweep by the max length of the sweep if there was not a full set of withdrawals
+        next_index = state.next_withdrawal_validator_index + MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP
+        next_validator_index = ValidatorIndex(next_index % len(state.validators))
+        state.next_withdrawal_validator_index = next_validator_index
+```
+
+#### New `verify_execution_payload_header_signature`
+
+```python
+def verify_execution_payload_header_signature(state: BeaconState, 
+                                           signed_header: SignedExecutionPayloadHeader) -> bool:
+    # Check the signature
+    builder = state.validators[signed_header.message.builder_index]
+    signing_root = compute_signing_root(signed_header.message, get_domain(state, DOMAIN_BEACON_BUILDER))
+    return bls.Verify(builder.pubkey, signing_root, signed_header.signature)
+```
+
+#### New `process_execution_payload_header`
+
+```python
+def process_execution_payload_header(state: BeaconState, block: BeaconBlock) -> None:
+    # Verify the header signature
+    signed_header = block.body.signed_execution_payload_header
+    assert verify_execution_payload_header_signature(state, signed_header)
+
+    # Check that the builder has funds to cover the bid
+    header = signed_header.message
+    builder_index = header.builder_index
+    amount = header.value
+    assert state.balances[builder_index] >= amount
+
+    # Verify that the bid is for the current slot
+    assert header.slot == block.slot
+    # Verify that the bid is for the right parent block
+    assert header.parent_block_hash == state.latest_block_hash
+    assert header.parent_block_root == block.parent_root
+
+    # Transfer the funds from the builder to the proposer
+    decrease_balance(state, builder_index, amount)
+    increase_balance(state, block.proposer_index, amount)
```

Comment:
TODO: We can use the churn mechanism of 7002 to transfer slowly these balances instead of lump transfer here.